src/module.c

```diff
@@ -6217,18 +6313,40 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
 
     user *user = NULL;
     if (flags & REDISMODULE_ARGV_RUN_AS_USER) {
-        user = ctx->user ? ctx->user->user : ctx->client->user;
+        if (ctx->client) {
+            user = ctx->client->user;
+        }
+        if (ctx->persistent_client) {
+            user = ctx->persistent_client->client->user;
+        } else if (ctx->user) {
+            user = ctx->user->user;
+        }
```

Comment:
i'm not certain that the priority should be like so.
maybe when `ctx->user` is explicitly set, we should prefer it?
could it be that `persistent_client->client->user` is not explicitly set and still we'll take the user from persistent_client?

Reply:
if there's no specific problem this combination creates, and just that we think it's odd, let's support it (flip the priority and add a test to cover it).

Reply:
done and added tests